electrum/plugins/jade/jadepy/jade.py

```diff
@@ -0,0 +1,656 @@
+import cbor
+import hashlib
+import json
+import time
+import logging
+import collections
+import collections.abc
+import traceback
+import random
+import sys
+
+
+# JadeError
+from .jade_error import JadeError
+
+# Low-level comms backends
+from .jade_serial import JadeSerialImpl
+from .jade_tcp import JadeTCPImpl
+
+# Not used in electrum wallet
+# Removed to reduce transitive dependencies
+# from .jade_ble import JadeBleImpl
+
+
+# Default serial connection
+DEFAULT_SERIAL_DEVICE = '/dev/ttyUSB0'
+DEFAULT_BAUD_RATE = 115200
+DEFAULT_SERIAL_TIMEOUT = 120
+
+# Default BLE connection
+DEFAULT_BLE_DEVICE_NAME = 'Jade'
+DEFAULT_BLE_SERIAL_NUMBER = None
+DEFAULT_BLE_SCAN_TIMEOUT = 60
+
+# 'jade' logger
+logger = logging.getLogger('jade')
+device_logger = logging.getLogger('jade-device')
+
+
+# Helper to map bytes-like types into hex-strings
+# to make for prettier message-logging
+def _hexlify(data):
+    if data is None:
+        return None
+    elif isinstance(data, bytes) or isinstance(data, bytearray):
+        return data.hex()
+    elif isinstance(data, list):
+        return [_hexlify(item) for item in data]
+    elif isinstance(data, dict):
+        return {k: _hexlify(v) for k, v in data.items()}
+    else:
+        return data
+
+
+# Simple http request function which can be used when a Jade response
+# requires an external http call.
+# The default implementation used in JadeAPI._jadeRpc() below.
+# NOTE: Only available if the 'requests' dependency is available.
+try:
+    import requests
```

Comment:
I've just realised that the PIN verification involves making HTTP network requests.
This is quite different from the other hwds where PIN verification happens on-device.
I assume this is intended. Do you have a high-level description somewhere how this works?

In any case, if the network requests are needed, the requests should respect the user's locally set proxy. Not sure how easy to do that is though; but could you take a look?

see e.g. `Network.send_http_on_proxy`:
https://github.com/spesmilo/electrum/blob/232e38e27dea91144ce4b2c057cc8db7de122936/electrum/network.py#L1320